fuse/plugins/micro_physics/lineage_cluster.py

```diff
@@ -372,18 +418,46 @@ def is_lineage_broken(
 ):
     """Function to check if the lineage is broken."""
 
+    if (
+        particle["creaproc"] == "RadioactiveDecayBase"
+        and particle["edproc"] == "RadioactiveDecayBase"
+    ):
+        # second step of a decay. We want to split the lineage
+        return True
+
     # In the nest code: Lineage is always broken if the parent is a ion
     # But if it's an alpha particle, we want to keep the lineage
-    break_for_ion = parent_lineage["lineage_type"] == 6
-    break_for_ion &= parent["type"] != "alpha"
-    break_for_ion &= particle["creaproc"] != "eIoni"
+    if (
+        (num_there(parent["type"]))
+        and ("[" not in parent["type"])
+        and (parent["parenttype"] == "none")
+        and (particle["type"] != "alpha")
+    ):
+        return True
 
-    if break_for_ion:
+    if (
+        (num_there(parent["type"]))
+        and ("[" not in parent["type"])
+        and (parent["creaproc"] == "RadioactiveDecayBase")
+        and (particle["type"] != "alpha")
+    ):
         return True
 
     # For gamma rays, check the distance between the parent and the particle
     if particle["type"] == "gamma":
 
+        if particle["creaproc"] == "phot" and particle["edproc"] == "phot":
+            # We do not want to split a photo absorption into two clusters
+            # The second photo absorption (that we see) could be x rays
+            # So, do not split if the distance is small
+            return False
```

Comment:
Here you do not evaluate the distance between the energy deposits? Please update the comment or code :-)